aten/src/ATen/core/qualified_name.h

```diff
@@ -1,91 +1,96 @@
 #pragma once
 
-#include <c10/util/intrusive_ptr.h>
+#include <c10/util/ArrayRef.h>
+#include <c10/util/Exception.h>
+#include <c10/util/StringUtil.h>
 #include <string>
 
 namespace c10 {
 
-// Represents names of the form, e.g., self.a.b
-struct QualifiedName;
-using QualifiedNamePtr = c10::intrusive_ptr<QualifiedName>;
-struct QualifiedName : c10::intrusive_ptr_target {
-  QualifiedName(QualifiedNamePtr prefix, std::string name)
-      : prefix_(std::move(prefix)), name_(std::move(name)) {
-    const auto pos = name_.find('.');
-    AT_ASSERTM(
-        pos == std::string::npos,
-        "Invalid name for qualified name: '",
-        name_,
-        "'");
-  }
-
-  const QualifiedNamePtr prefix_ = QualifiedNamePtr();
-  const std::string name_;
-
-  static QualifiedNamePtr create(QualifiedNamePtr prefix, std::string name) {
-    return c10::make_intrusive<QualifiedName>(
-        std::move(prefix), std::move(name));
-  }
-  static QualifiedNamePtr create(std::string name) {
-    return c10::make_intrusive<QualifiedName>(
-        QualifiedNamePtr(), std::move(name));
-  }
+// Represents a name of the form "foo.bar.baz"
+struct QualifiedName {
```

Comment:
Question: why remove QualifiedNamePtr?

Reply:
I explain the reasoning in the PR description